src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionBundleValidator.cs

```diff
@@ -68,50 +70,47 @@ private static string BuildRequestUrlForConditionalQueries(EntryComponent entry,
             return string.IsNullOrWhiteSpace(conditionalCreateQuery) ? entry.Request.Url : entry.Request.Url + "?" + conditionalCreateQuery;
         }
 
-        private async Task<string> GetResourceId(EntryComponent entry, CancellationToken cancellationToken)
+        private async Task<string> GetResourceId(EntryComponent entry, IDictionary<string, (string resourceId, string resourceType)> idDictionary, CancellationToken cancellationToken)
         {
+            // If there is no search or conditional operations, then use the FullUrl for posts and the request url otherwise
             if (string.IsNullOrWhiteSpace(entry.Request.IfNoneExist) && !entry.Request.Url.Contains("?", StringComparison.Ordinal))
             {
-                if (entry.Request.Method == HTTPVerb.POST)
-                {
-                    return entry.FullUrl;
-                }
+                return entry.Request.Method == HTTPVerb.POST ? entry.FullUrl : entry.Request.Url;
+            }
+
+            string resourceType = null;
+            StringValues conditionalQueries;
 
-                return entry.Request.Url;
+            if (entry.Request.Method == HTTPVerb.PUT)
+            {
+                string[] conditionalUpdate = entry.Request.Url.Split('?');
```

Comment:
not your code but I feel like we shold use existing library to parse URLs instead of implementing this ourselves. For example, I think there is a bug here if one of the query parameter contains value ? (if this were truly URL, additional ? will be encoded as %3F but because this is just a JSON string, i am not sure if it will always be encoded.) If there is additional ? in the value without encoded, then the conditionalQueries below might be wrong?